.scripts/generate-sdks.ts

```diff
@@ -140,35 +159,102 @@ export async function findMissingSdks(azureRestApiSpecsRepository: string): Prom
     return missingSdks;
 }
 
-async function doesReadmeMdFileSpecifiesTypescriptSdk(readmeMdPath: string): Promise<boolean> {
-    const readmeMdBuffer = await fs.readFile(readmeMdPath);
-    const sectionBeginning = "``` yaml $(swagger-to-sdk)"
-    const sectionEnd = "```"
-
-    const beginningIndex = readmeMdBuffer.indexOf(sectionBeginning);
-    const trimmedBuffer = readmeMdBuffer.slice(beginningIndex);
+async function getYamlSection(buffer: Buffer, sectionBeginning: string, sectionEnd: string): Promise<Buffer> {
+    const beginningIndex = buffer.indexOf(sectionBeginning);
+    const trimmedBuffer = buffer.slice(beginningIndex + (sectionBeginning.length));
 
-    const endIndex = trimmedBuffer.indexOf(sectionEnd);
+    const endIndex = trimmedBuffer.indexOf(sectionEnd, 3);
     const sectionBuffer = trimmedBuffer.slice(0, endIndex);
 
+    return sectionBuffer;
+}
+
+async function doesReadmeMdFileSpecifiesTypescriptSdk(readmeMdPath: string): Promise<boolean> {
+    const readmeMdBuffer = await fs.readFile(readmeMdPath);
+    const sectionBuffer = await getYamlSection(readmeMdBuffer, "``` yaml $(swagger-to-sdk)", "```");
+
     if (sectionBuffer.includes("azure-sdk-for-js")) {
         return true;
     }
 
     return false;
 }
 
-export async function copyExistingNodeJsReadme(sdkPath: string): Promise<void> {
+export async function copyExistingNodeJsReadme(sdkPath: string): Promise<string> {
     const nodeJsReadmePath = path.resolve(sdkPath, "readme.nodejs.md");
     const typescriptReadmePath = path.resolve(sdkPath, "readme.typescript.md");
 
     if (args.verbose) {
-        console.log(`Copying ${nodeJsReadmePath} to ${typescriptReadmePath}`)
+        _logger.log(`Copying ${nodeJsReadmePath} to ${typescriptReadmePath}`)
     }
 
     if (await exists(typescriptReadmePath)) {
         throw new Error(`${typescriptReadmePath} file already exists`)
     }
 
     await fs.copyFile(nodeJsReadmePath, typescriptReadmePath);
+    return typescriptReadmePath;
+}
+
+async function updatePackageName(settings: readmeSettings): Promise<readmeSettings> {
+    const packageName = settings.nodejs["package-name"]
+    if (packageName.startsWith("arm")) {
+        return settings;
+    }
+
+    if (!packageName.startsWith("azure-")) {
+        throw new Error("Unrecognized package name. Please update manually.");
```

Comment:
In this scenario we'd actually probably just take the name as it is. Consider the `keyvault` data-plane package. In our new SDK it will still be `keyvault`, even though it's management-plane package changed from `azure-arm-keyvault` to `arm-keyvault`.